packages/storybook/src/executors/utils.ts

```diff
@@ -302,6 +305,40 @@ export function isStorybookLT6() {
   );
 }
 
+export function customProjectBuildConfigIsValid(
+  tree: Tree,
+  projectBuildConfig: string
+): boolean {
+  if (projectBuildConfig.includes(':')) {
+    const { project, target } = parseTargetString(projectBuildConfig);
+    const projectConfig = readProjectConfiguration(tree, project);
+    if (projectConfig?.targets?.[target]) {
+      return true;
+    } else {
+      logger.warn(`
+      The projectBuildConfig you provided is not valid.   
+      ${!projectConfig ? 'The project ' + project + ' does not exist.' : ''}   
+      ${
+        projectConfig &&
+        !projectConfig.targets?.[target] &&
+        `The project ${project} does not have the target ${target}.`
+      }  
+      The default projectBuildConfig is going to be used.       
+      `);
+    }
+  } else {
+    try {
+      return !!readProjectConfiguration(tree, projectBuildConfig);
```

Comment:
nit:
```suggestion
      return Boolean(readProjectConfiguration(tree, projectBuildConfig));
```

I feel like this method of coercing a boolean value shows more intent. But it is up to you :)